command-expansion-server/src/lib/batchLoaders/expansionSetBatchLoader.ts

```diff
@@ -1,35 +1,75 @@
 import { BatchLoader } from './index';
 import { gql, GraphQLClient } from 'graphql-request';
 import { ErrorWithStatusCode } from '../../utils/ErrorWithStatusCode.js';
+import { GraphQLActivitySchema } from './activitySchemaBatchLoader.js';
 
-// export const expansionSetBatchLoader: BatchLoader<
-//   { expansionSetId: number },
-//   ExpansionSet,
-//   { graphqlClient: GraphQLClient }
-// > = opts => async keys => {
-//   const {simulation_dataset} = await opts.graphqlClient.request<{
-//     expansion_sets
-//   }>(gql`
-//     query GetSimulationDatasetActivities($simulationDatasetIds: [Int!]!) {
-//       expansion
-//     }
-//   `, {
-//     simulationDatasetIds: keys.map(key => key.simulationDatasetId),
-//   });
-//
-//   return Promise.all(keys.map(async ({ simulationDatasetId }) => {
-//     const simulationDataset = simulation_dataset.find(simulationDataset => simulationDataset.id === simulationDatasetId);
-//     if (simulationDataset === undefined) {
-//       return new ErrorWithStatusCode(`No simulation_dataset with id: ${simulationDatasetId}`, 404);
-//     }
-//     return simulationDataset.dataset.spans;
-//   }));
-// };
-//
-// export interface ExpansionSet {
-//   commandDictionaryId: number,
-//   missionModelId: number,
-//   expansionIds: number[],
-//   // TODO: replace with actual expansions
-// }
+export const expansionSetBatchLoader: BatchLoader<
+  { expansionSetId: number },
+  ExpansionSet,
+  { graphqlClient: GraphQLClient }
+> = opts => async keys => {
+  const {expansion_set} = await opts.graphqlClient.request<{
+    expansion_set: {
+      id: number
+      command_dictionary: {
+        id: number,
+        command_types_typescript_path: string
+      },
+      mission_model: {
+        activity_type: (GraphQLActivitySchema & { model_id: number })[]
+      },
+      expansion_rules: {
+        id: number,
+        activity_type: string,
+        expansion_logic: string,
+      }
+    }[]
+  }>(gql`
+    query GetExpansionSet($expansionSetIds: [Int!]!) {
+      expansion_set(where: { id: {_in: $expansionSetIds } }) {
+        id
+        command_dictionary {
+          id
+          command_types_typescript_path
+        }
+        mission_model {
+          activity_type {
+            name
+            parameters
+            computed_attributes_value_schema
+            model_id
+          }
+        }
+        expansion_rules {
+          id
+          activity_type
+          expansion_logic
+        }
+      }
+    }
+  `, {
+    simulationDatasetIds: keys.map(key => key.expansionSetId),
+  });
 
+  return Promise.all(keys.map(async ({ expansionSetId }) => {
+    const expansionSet = expansion_set.find(table => table.id === expansionSetId);
+    if (expansionSet === undefined) {
+      return new ErrorWithStatusCode(`No simulation_dataset with id: ${expansionSetId}`, 404);
+    }
+    return expansionSet;
```

Comment:
We should pull the typescript types for the command dictionary from the filesystem in here as well so we have those for expansion

Reply:
Latest commits should resolve this.